.github/actions/setup-job-macos/action.yml

```diff
@@ -0,0 +1,94 @@
+name: 'Setup job for self-hosted macOS runner'
+description: 'Setup rust and flutter'
+inputs:
+  flutter:
+    description: 'Setup flutter'
+    default: "false"
+    required: false
+  rust-target:
+    description: 'Install an additional rust target'
+    default: ""
+    required: false
+  cache-key:
+    description: 'An optional key that is added to the automatic cache key'
+    default: ""
+    required: false
+  github_token:
+    description: 'The github token used for the `just` action to avoid exceeding API rate limits'
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Clean up runner tool cache and temp dir
+      shell: bash
+      run: |
+        cd $RUNNER_TOOL_CACHE
```

Comment:
🔧  We should verify here that `RUNNER_TOOL_CACHE` and `RUNNER_TEMP` aren't empty. If they were empty for whatever reason, we'd be deleting the entire home directory of the M1 Mac

🤔 Optimally we would also check whether they are subdirectories of the runner main directory.

Reply:
I haven't found in the documentation whether the string can be empty or whether it is never empty :/
> If they were empty for whatever reason, we'd be deleting the entire home directory of the M1 Mac

In this case I think we should tweak the runner's permissions on the host so if this happens it will fail due to insufficient permissions. wdyt?

For know I have added a check that the last dir name is `_tool` or `_temp`. Do you think that is enough?

Reply:
It could also be empty due to a bug or an error on our side, so even if the documentation says no, it would make sense to check it I think. 

You're right in that more granular permissions would be nice, and we could look into doing that. However, I fear it might not be as simple as it'd be on linux. At least I have bad experiences with multi-user macOS, so maybe that's just why I'm skeptical 😅

It just occurred to me that, for good measure, we could throw in `set -euo pipefail` ([bash 'safe mode'](http://redsymbol.net/articles/unofficial-bash-strict-mode/)). That should catch problems. Sorry this occurred to me a little late :D

Reply:
Good idea 👍 i have added the `-u` option. `-eo pipefail` are already set by the runner https://github.com/xaynetwork/xayn_discovery_engine/runs/6025070446?check_suite_focus=true#step:4:30